### PR TITLE
Implement LayerFactory

### PR DESCRIPTION
**Pull request by chrismayer**
This adds a LayerFactory singleton to create OL layers from a JSON configuration. Supported layers at the moment:

  - OSM
  - WMS
  - ESRI REST

### CODE REVIEW

File: app/factory/Layer.js [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+
+        var layerType = layerConf.layerType;
+        var mapLayer;
+
+        switch (layerType) {
+        case 'wms':
+            mapLayer = LayerFactory.createWms(layerConf);
+            break;
+        case 'wfs':
+            mapLayer = LayerFactory.createWFS(layerConf);
```
Comment (marcjansen) [inline app/factory/Layer.js]: We should create an issue wrt to handling abbreviations and casing consistently. Non-blocking of course.

File: app/factory/Layer.js [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+            break;
+        case 'google_terrain':
+            mapLayer = LayerFactory.createGoogle(layerConf, 'google.maps.MapTypeId.TERRAIN');
+            break;
+        case 'google_hybrid':
+            mapLayer = LayerFactory.createGoogle(layerConf, 'google.maps.MapTypeId.HYBRID');
+            break;
+        case 'google_satellite':
+            mapLayer = LayerFactory.createGoogle(layerConf, 'google.maps.MapTypeId.SATELLITE');
+            break;
```
Comment (marcjansen) [inline app/factory/Layer.js]: We either need a 3rd party plugin for these or remove them altogether. Lets create an issue

File: app/factory/Layer.js [hunk 3]
```diff
@@ -0,0 +1,184 @@
...
+            Ext.log.warn('Layer type not known');
+            //do nothing, and return empty layer
+        }
+
+        return mapLayer;
+    },
+
+    createEmptyLayer: function(layerConf) {
+        Ext.log.info('Not implemented yet', layerConf);
+    },
```
Comment (marcjansen) [inline app/factory/Layer.js]: We should implement this, so more complex configs worl out of the box.

File: app/factory/Layer.js [hunk 4]
```diff
@@ -0,0 +1,184 @@
...
+            break;
+        case 'arcgisrest':
+            mapLayer = LayerFactory.createArcGISRest(layerConf);
+            break;
+        case 'switchlayer':
+            mapLayer = LayerFactory.createSwitchLayer(layerConf);
+            break;
+        default:
+            Ext.log.warn('Layer type not known');
+            //do nothing, and return empty layer
```
Comment (marcjansen) [inline app/factory/Layer.js]: Call to createEmptyLayer missing

File: app/factory/Layer.js [hunk 5]
```diff
@@ -0,0 +1,184 @@
...
+    /**
+     * Creates an OGC WMS layer
+     *
+     * @param  {Object} layerConf The configuration object for this layer
+     * @return {ol.layer.Tile}    WMS layer
+     */
+    createWms: function(layerConf) {
+        return new ol.layer.Tile({
+            name: layerConf.text,
+            source: new ol.source.TileWMS({
```
Comment (marcjansen) [inline app/factory/Layer.js]: We need a nissue here wrt to having this configurable probably.

File: app/factory/Layer.js (hunk 1 above)
Comment (chrismayer) [reply app/factory/Layer.js]: Agree, this was transferred by some legacy code.

File: app/factory/Layer.js (hunk 2 above)
Comment (chrismayer) [reply app/factory/Layer.js]: See #8

File: app/factory/Layer.js (hunk 3 above)
Comment (chrismayer) [reply app/factory/Layer.js]: They should because the addition of the layers checks if we have valid layer instances

**Pull request merged**